### PR TITLE
use rcu in DiskManager

### PR DESCRIPTION
**Pull request by liwenhui-soul**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [ ] enhancement

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: src/kvstore/DiskManager.h [hunk 1]
```diff
@@ -90,37 +93,41 @@ class DiskManager {
...
-  // given a space and data path, return all parts in the path
-  std::unordered_map<GraphSpaceID, PartDiskMap> partPath_;
-
-  // the index in dataPaths_ for a given space + part
-  std::unordered_map<GraphSpaceID, std::unordered_map<PartitionID, size_t>> partIndex_;
-
   // lock used to protect partPath_ and partIndex_
   std::mutex lock_;
+
+  mutable folly::rcu_domain<> rcuDomain_;
```
Comment (critical27) [inline src/kvstore/DiskManager.h:130]: Maybe you don't need to declare a `rcu_domain` explicitly, the default one is enough.

File: src/kvstore/DiskManager.h [hunk 2]
```diff
@@ -90,37 +93,41 @@ class DiskManager {
...
   std::vector<std::atomic_uint64_t> freeBytes_;
 
-  // given a space and data path, return all parts in the path
-  std::unordered_map<GraphSpaceID, PartDiskMap> partPath_;
-
-  // the index in dataPaths_ for a given space + part
-  std::unordered_map<GraphSpaceID, std::unordered_map<PartitionID, size_t>> partIndex_;
-
   // lock used to protect partPath_ and partIndex_
   std::mutex lock_;
```
Comment (critical27) [inline src/kvstore/DiskManager.h:128]: I think we don't need the mutex now?

File: src/kvstore/DiskManager.h (hunk 1 above)
Comment (liwenhui-soul) [reply src/kvstore/DiskManager.h:130]: > Maybe you don't need to declare a `rcu_domain` explicitly, the default one is enough.

metaclient also use default, they share the same one?

File: src/kvstore/DiskManager.h (hunk 2 above)
Comment (liwenhui-soul) [reply src/kvstore/DiskManager.h:128]: > I think we don't need the mutex now?

write & write need it

Comment (critical27) [reply src/kvstore/DiskManager.h:128]: 👌 , got it

**Review by critical27 — APPROVED** [review]
Good job~

**Review by cangfengzhs — APPROVED** [review]
LGTM

**Pull request merged**